data-service/src/main/java/com/amazonaws/blox/dataservice/handler/EnvironmentHandler.java

```diff
@@ -30,26 +31,71 @@
 
   @NonNull private EnvironmentRepository environmentRepository;
 
-  public Environment createEnvironment(final String environmentName, final String accountId)
-      throws EnvironmentExistsException, StorageException {
-    return null;
+  public Environment createEnvironment(final Environment environment)
+      throws EnvironmentExistsException, ServiceException {
+    try {
+      return environmentRepository.createEnvironment(environment);
+    } catch (final StorageException e) {
+      throw new ServiceException(
+          String.format(
+              "Exception occurred when creating environment with id %s and version %s",
+              environment.getEnvironmentId(), environment.getEnvironmentVersion()),
+          e);
+    }
   }
 
-  public Environment describeEnvironment(final String environmentId) {
-    return null;
+  public Environment createEnvironmentTargetVersion(final Environment environment)
+      throws EnvironmentExistsException, ServiceException {
+    try {
+      return environmentRepository.createEnvironmentTargetVersion(environment);
```

Comment:
Tracing this through, it looks like this makes it possible for callers to set totally different values on the `environment` being passed in than what is actually in the `Environment` table. Should this method perhaps only take `(environmentId, environmentVersion)` as input?

Reply:
yeah you're right. and now that I think about it we don't want them to set it on createEnvironment either: things like type and health etc. I'll change it